src/test/java/unit/AuthLoginLogout.java

```diff
@@ -38,14 +39,26 @@ void loginTestUser() throws IOException {
     testUser.login();
 
     // check that the credential exists in the store on disk
-    DataStore<StoredCredential> dataStore = TestUser.getCredentialStore();
-    assertEquals(1, dataStore.keySet().size(), "credential store only contains one entry");
+    DataStore<StoredCredential> credentialStore = TestUser.getCredentialStore();
+    assertEquals(2, credentialStore.keySet().size(), "credential store contains two entries");
+    assertTrue(
+        credentialStore.containsKey(GoogleOauth.CREDENTIAL_STORE_KEY),
+        "credential store contains hard-coded user key");
     assertTrue(
-        dataStore.containsKey(GoogleOauth.CREDENTIAL_STORE_KEY),
-        "credential store contains hard-coded single user key");
-    StoredCredential storedCredential = dataStore.get(GoogleOauth.CREDENTIAL_STORE_KEY);
+        credentialStore.containsKey(GoogleOauth.ID_TOKEN_STORE_KEY),
+        "credential store contains  id token");
+    StoredCredential storedCredential = credentialStore.get(GoogleOauth.CREDENTIAL_STORE_KEY);
     assertThat(storedCredential.getAccessToken(), CoreMatchers.not(emptyOrNullString()));
 
+    DataStore<IdToken> idTokenStore = TestUser.getCredentialStore();
+    assertEquals(
+        idTokenStore.keySet(),
+        credentialStore.keySet(),
+        "credentialStore and idTokenStore are different views of same underlying datastore");
```

Comment:
Are you asserting an implementation detail here? I.e. could you have just as easily implemented it with two different data stores?

Reply:
I'm not completely sure what the intent of this test is, but I tried to match it.  I did very intentionally put the ID token into the same "DataStore" file as the GoogleCredential, under different keys, rather than having them exist in separate files.  

Side note, the semantics of `DataStore<T>` are very odd to me, in that it uses the parameterized type to give a view of an entry in a file data store that may contain different keyed entries of different types.  Seems like it should be something like `DataStoreEntry<T>` 🤷

Reply:
Yeah it was the first time I'd seen that pattern.